File: discipline_bridge/conditionals/time_conditional.ts
import { TypeId, Weekday, Time, WeekdaySet, TimeRange, Unique } from "../mod.ts";

export class TimeConditional implements Unique {
  readonly typeId = TypeId.TimeConditional;

  private constructor(
    private readonly _weekdays: WeekdaySet,
    private readonly _timeRange: TimeRange
  ) {}

  isEffective(weekday: Weekday, time: Time): boolean {
    return this._weekdays.contains(weekday) && this._timeRange.contains(time)
  }
}

File: discipline_daemon/src/conditionals/time_conditional.rs
use serde::{Deserialize, Serialize};
use crate::x::{Time, TimeRange, Weekday, WeekdaySet};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TimeConditional {
  time_range: TimeRange,
  weekday_set: WeekdaySet,
}

impl TimeConditional {
  pub fn new(time_range: TimeRange, weekday_set: WeekdaySet) -> Self {
    Self {
      time_range,
      weekday_set,
    }
  }

  pub fn evaulate(&self, time: Time, weekday: Weekday) -> bool {
    self.time_range.contains(time)
    &&
    self.weekday_set.contains(weekday)
  }
}